src/megameklab/com/ui/view/CVTransportView.java

```diff
@@ -0,0 +1,220 @@
+import megamek.common.verifier.BayData;
+import megameklab.com.ui.view.listeners.CVBuildListener;
+
+/**
+ * Panel for combat vehicle cargo and troop space.
+ * 
+ * @author Neoancient
+ *
+ */
+public class CVTransportView extends BuildView implements ChangeListener {
+    
+    /**
+     * 
+     */
+    private static final long serialVersionUID = -3059498307570586741L;
+    
+    List<CVBuildListener> listeners = new CopyOnWriteArrayList<>();
+    public void addListener(CVBuildListener l) {
+        listeners.add(l);
+    }
+    public void removeListener(CVBuildListener l) {
+        listeners.remove(l);
+    }
+
+    private final SpinnerNumberModel spnFixedTroopModel = new SpinnerNumberModel(0.0, 0.0, null, 0.5);
+    private final SpinnerNumberModel spnPodTroopModel = new SpinnerNumberModel(0.0, 0.0, null, 0.5);
+    private final Map<BayData, SpinnerNumberModel> fixedSpinnerModels = new HashMap<>();
+    private final Map<BayData, SpinnerNumberModel> podSpinnerModels = new HashMap<>();
+    
+    private final JSpinner spnFixedTroop = new JSpinner(spnFixedTroopModel);
+    private final JSpinner spnPodTroop = new JSpinner(spnPodTroopModel);
+    private final Map<BayData, JSpinner> fixedSpinners = new HashMap<>();
+    private final Map<BayData, JSpinner> podSpinners = new HashMap<>();
+    
+    // Track unit tonnage to set max allowed carrying space.
+    private double tonnage;
+
+    public CVTransportView() {
+        super();
+        initUI();
+    }
+    
+    private void initUI() {
+        ResourceBundle resourceMap = ResourceBundle.getBundle("megameklab.resources.Views", new EncodeControl()); //$NON-NLS-1$
+        
+        setLayout(new GridBagLayout());
+        GridBagConstraints gbc = new GridBagConstraints();
+
+        gbc.gridx = 1;
+        gbc.gridy = 0;
+        gbc.fill = java.awt.GridBagConstraints.HORIZONTAL;
+        gbc.anchor = GridBagConstraints.WEST;
+        add(createLabel(resourceMap.getString("CVTransportView.lblFixed.text"), labelSize), gbc); //$NON-NLS-1$
+        gbc.gridx = 2;
+        gbc.gridy = 0;
+        add(createLabel(resourceMap.getString("CVTransportView.lblPod.text"), labelSize), gbc); //$NON-NLS-1$
+
+        gbc.gridx = 0;
+        gbc.gridy = 1;
+        add(createLabel(resourceMap.getString("CVTransportView.lblTroopSpace.text"), labelSizeLg), gbc); //$NON-NLS-1$
+        
+        gbc.gridx = 1;
+        setFieldSize(spnFixedTroop, editorSize);
+        add(spnFixedTroop, gbc);
+        spnFixedTroop.addChangeListener(this);
+        
+        gbc.gridx = 2;
+        setFieldSize(spnPodTroop, editorSize);
+        add(spnPodTroop, gbc);
+        spnPodTroop.addChangeListener(this);
+        
+        for (BayData bayType : BayData.values()) {
+            if (!bayType.isCargoBay()) {
+                continue;
+            }
+            String tooltip = String.format(resourceMap.getString("CVTransportView.bay.tooltipFormat"),
+                    1 / bayType.getWeight());
```

Comment:
The tonnage devoted to cargo space. Each ton [of what?] provides [1 / bay tonnage?] tons of cargo capacity.

I assume it's each ton of bay, but I have no idea what the 1 / bay tonnage figure is supposed to be.

Reply:
bayType.getWeight() returns the tonnage of a single unit of the bay (one cubicle or one ton of cargo capacity). The reciprocal is the cargo capacity per ton of bay. I just added the missing javadoc comments to `BayData` in MM.